src/mobu/services/business/notebookrunner.py

```diff
@@ -158,6 +170,12 @@ async def execute_code(self, session: JupyterLabSession) -> None:
             self.logger.info(msg)
 
             for cell in self.read_notebook(self._notebook):
+                if self.refreshing:
+                    await self.refresh()
+                    self.logger.info(
+                        "Recloning notebooks and forcing new execution"
+                    )
```

Comment:
I would put this log message in the `refresh` method since I think you always want this message even if refresh is called a different way.